engines/dfc_provider/spec/services/supplied_product_builder_spec.rb

```diff
@@ -79,6 +84,14 @@
 
       expect(product.image).to eq variant.product.image.url(:product)
     end
+
+    it "assigns the product uri" do
+      product = builder.supplied_product(variant)
+
+      expect(product.spree_product_uri).to eq(
+        "http://test.host/api/dfc/enterprises/7?spree_product_id=6"
```

Comment:
It's just the enterprise URL with the spree_product_id as parameter. This can be given as hash to the URL helper:
https://github.com/openfoodfoundation/openfoodnetwork/blob/29dad44f9fd181d4cdd7ff109b40f9c8d0985d77/engines/dfc_provider/app/services/supplied_product_builder.rb#L9-L12